google/cloud/aiplatform/training_jobs.py

```diff
@@ -15,9 +15,297 @@
+    """
+
+    python_executable = sys.executable
+
+    if not python_executable:
+        raise EnvironmentError("Cannot find Python executable for packaging.")
+    return python_executable
+
+
+class _TrainingScriptPythonPackager:
+    """Converts a Python script into Python package suitable for aiplatform training.
+
+    Copies the script to specified location.
+
+    Class Attributes:
+        _TRAINER_FOLDER: Constant folder name to build package.
+        _ROOT_MODULE: Constant root name of module.
+        _TEST_MODULE_NAME: Constant name of module that will store script.
+        _SETUP_PY_VERSION: Constant version of this created python package.
+        _SETUP_PY_TEMPLATE: Constant template used to generate setup.py file.
+        _SETUP_PY_SOURCE_DISTRIBUTION_CMD:
+            Constant command to generate the source distribution package.
+
+    Attributes:
+        script_path: local path of script to package
+        requirements: list of Python dependencies to add to package
+
+    Usage:
+
+    packager = TrainingScriptPythonPackager('my_script.py', ['pandas', 'pytorch'])
+    gcs_path = packager.package_and_copy_to_gcs(
+        staging_bucket='my-bucket',
+        project='my-prject')
+    module_name = packager.module_name
+
+    The package after installed can be executed as:
+    python -m aiplatform_custom_trainer_script.task
+
+    """
+
+    _TRAINER_FOLDER = "trainer"
+    _ROOT_MODULE = "aiplatform_custom_trainer_script"
+    _TASK_MODULE_NAME = "task"
+    _SETUP_PY_VERSION = "0.1"
+
+    _SETUP_PY_TEMPLATE = """from setuptools import find_packages
+from setuptools import setup
+
+setup(
+    name='{name}',
+    version='{version}',
+    packages=find_packages(),
+    install_requires=({requirements}),
+    include_package_data=True,
+    description='My training application.'
+)"""
+
+    _SETUP_PY_SOURCE_DISTRIBUTION_CMD = (
+        "{python_executable} setup.py sdist --formats=gztar"
+    )
+
+    def __init__(self, script_path: str, requirements: Optional[Sequence[str]] = None):
+        """Initializes packager.
+
+        Args:
+            script_path (str): Required. Local path to script.
+            requirements (Sequence[str]):
+                List of python packages dependencies of script.
+        """
+
+        self.script_path = script_path
+        self.requirements = requirements or []
+
+    def make_package(self, package_directory: str) -> str:
+        """Converts script into a Python package suitable for python module execution.
+
+        Args:
+            package_directory (str): Directory to build package in.
+        Returns:
+            source_distribution_path (str): Path to built package.
+        Raises:
+            RunTimeError if package creation fails.
+        """
+        # The root folder to builder the package in
+        package_path = pathlib.Path(package_directory)
+
+        # Root directory of the package
+        trainer_root_path = package_path / self._TRAINER_FOLDER
```

Comment:
This would be writing into the user's file system and could potentially overwrite existing files.  Perhaps build the package in a tmp folder instead?

Reply:
This method is invoked through `package_and_copy` which passes in a tmp folder for this reason.

This is a private class so we don't intend for the user to use this class directly.